src/profiles/version.rs

```diff
@@ -0,0 +1,181 @@
+//! # Profile API Versioning.
+//!
+//! Profile API versioning is identified using X.Y format (e.g: 3.4).
+//!
+//! ## Version semantics
+//! A change in the minor (Y) digit of the version indicates that the new Profile API adds some
+//! nice-to-have functionality that:
+//! 1) Does not break yaml parsing, i.e: a retis version that is able to parse a profile version
+//!    X.Y should also be able to parse X.(Y+1)
+//! 2) Adds some functionality that is nice to have but not absolutely mandatory for the profile to
+//!    yield a consistent and working configuration. A retis version that supports version X.1 will
+//!    try to load and execute a profile with version X.Y and a reasonably reasonably good result
+//!    is expected (i.e: the tool must not fail and events should be captured).
+//!
+//! On the other hane, a change in the mayor (X) digit of the version indicates that the change can
+//! break parsing or, even if it doesn't, the change in behavior can make the tool fail if not
+//! applied by a retis version that supports the new functionality.
+//!
+//! ## Backwards compatibility
+//! Retis only guarantees backwards compatibility within the same major version.
+
+use core::{fmt, str::FromStr};
+use std::cmp::Ordering;
+
+use anyhow::{anyhow, bail, Error, Result};
+use serde::{de::Error as Derror, Deserialize, Deserializer};
+
+/// A simple two-digit version struct implementing Profile API Version schema and support logic.
+/// No VersionReq is implemented. Instead, requirement satisfaction is implemented manually.
+#[derive(Debug, PartialEq)]
+pub(crate) struct ApiVersion {
+    pub(super) major: u32,
+    pub(super) minor: u32,
+}
+
+/// Possible Version compatibility results.
+#[derive(Debug, PartialEq)]
+pub(super) enum ApiVersionSupport {
+    Full,
+    Partial,
+    NotSupported,
+}
+
+impl ApiVersion {
+    /// Create `ApiVersion` by parsing from string representation.
+    pub(crate) fn parse(text: &str) -> Result<Self> {
+        ApiVersion::from_str(text)
+    }
+
+    /// Assuming `self` is retis' supported API version, return whether parsing the ApiVersion
+    /// referenced by `profile` is supported.
+    pub(super) fn supports(&self, profile: &ApiVersion) -> Result<ApiVersionSupport> {
+        if self.major == profile.major {
+            if self.minor >= profile.minor {
+                Ok(ApiVersionSupport::Full)
+            } else {
+                Ok(ApiVersionSupport::Partial)
+            }
+        } else {
+            Ok(ApiVersionSupport::NotSupported)
```

Comment:
The whole point of adding a very basic compatibility specification is to avoid having to write a per-field and per-version compatibility check, right? At what level would you verify compatibility without implementing such a big compatibility matrix?